EIPS/eip-1884.md

```diff
@@ -0,0 +1,146 @@
+
+![graph](../assets/eip-1884/SLOAD-run3.png)
+
+It can be seen that the repricing at [EIP-150][eip-150] caused a steep drop, from around `67` to `23`. 
+Around block `5M`, it started reaching pre-[EIP-150][eip-150] levels, and at block `7M` 
+it was averaging on around `150` - more than double pre-eip-150 levels. 
+
+Increasing the cost of `SLOAD` by `4` would bring it back down to around `40`. 
+It is to be expected that it will rise again in the future, and may need future repricing, unless 
+state clearing efforts are implemented before that happens. 
+
+### `BALANCE` 
+
+`BALANCE` (a.k.a `EXTBALANCE`) is an operation which fetches data from the state trie. It was repriced at [EIP-150][eip-150] from `20` to `400`.
+
+![graph](../assets/eip-1884/BALANCE-run3.png)
+
+It is comparable to `EXTCODESIZE` and `EXTCODEHASH`, which are priced at `700` already. 
+
+It has a built-in variance, since it is often used for checking the balance of `this`, 
+which is a cheap operation, however, it can be used to lookup the balance of arbitrary account. 
+
+In hindsight, it might have been a better choice to have two 
+opcodes: `EXTBALANCE(address)` and `SELFBALANCE`, and have two different prices. 
+
+* Specification (A) does not extend the current opcode set, thus the safest alternative is to set the price at the worst-case basis.
+* Specification (B) extends the current opcode set.
+  * Unfortunately, the opcode span `0x3X` is already full, hence the suggestion to place `SELFBALANCE` in the `0x4X` range.  
+  * As for why it is priced at `5` (`GasFastStep`) instead of `2` (`GasQuickStep`), like other similar operations: the EVM execution engine still needs a lookup into the (cached) trie, and `balance`, unlike `gasPrice` or `timeStamp`, is not constant during the execution, so it has a bit more inherent overhead. 
+
+
+
+## Backwards Compatibility
+
+The changes require a hardfork. The changes have the following consequences:
+
+- Certain calls will become more expensive.
+- Contracts that assume a certain fixed gas cost for calls (or internal sections) may cease to function.
+  - However, these operations have already been repriced earlier, so there is a historical precedent that 'the gascost for these operations may change', which should have prevented such fixed-gas-cost assumptions from being implemented.
+
+I expect that certain patterns will be less used, for example the use of multiple modifiers which `SLOAD`s the same opcode will be merged into one. It may also lead to less `log` operations containing `SLOAD`ed values that are not strictly necessary.
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Other EIPs can choose to include links to test cases if applicable.-->
+
+No test cases are implemented as of yet.
+
+## Implementation
+
+Both these opcodes have been repriced before, and the client internals for managing reprices are already in place.
```

Comment:
Apart from a technical specification, it is the place to link to clients that have implemented the change.